presto-main/src/main/java/com/facebook/presto/cost/SampleStatsRule.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.cost.ComposableStatsCalculator.Rule;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.SampleNode;
+
+import java.util.Optional;
+
+import static com.facebook.presto.sql.planner.plan.Patterns.sample;
+
+public class SampleStatsRule
+        implements Rule<SampleNode>
```

Comment:
extend SimpleStatsRule instead so the stats get normalized

Reply:
Why do some rules need normalization but others don't?

Reply:
Thanks @rschlussel  for your help and review. I have amended the change now. I have similar question as @aweisberg . Can you help in terms of when normalization is required?

Reply:
If you're only adjusting one field from stats but not others, you'd want to normalize. Normalization makes it so that e.g. if you set the row count less than the distinct value count, we reduce the distinct value count to match.  

Almost all rules extend SimpleStatsRule right now. I think it would be quite reasonable to require it for every rule, or even to do it in the PlanNodeStatsEstimate builder or constructor, but that's beyond the scope here.

Reply:
IMO implicit non-obvious knowledge like this should be documented somewhere. We can't rely on having the right experts around to safely make changes to this code.

Reply:
Agreed, though I think instead we should make it happen automatically by default, so that it's impossible to mess it up.